grovedb/src/lib.rs

```diff
@@ -461,11 +460,31 @@ impl GroveDb {
         self.subtrees = self.temp_subtrees.drain().collect();
 
         // TODO: root tree actually does support transactions, so this
-        // code can be reworked to account for that
+        //  code can be reworked to account for that
+
+        // Free transactional data
         self.temp_root_tree = MerkleTree::new();
+        self.temp_root_leaf_keys = HashMap::new();
+        self.temp_subtrees = HashMap::new();
 
         Ok(db_transaction
             .commit()
             .map_err(PrefixedRocksDbStorageError::RocksDbError)?)
     }
+
+    /// Rollbacks previously started db transaction. For more details on the
+    /// transaction usage, please check [`GroveDb::start_transaction`]
+    pub fn rollback_transaction(
+        &mut self,
+        db_transaction: &OptimisticTransactionDBTransaction,
```

Comment:
Like `commit_transaction` this method should take ownership of `db_transaction`, because it is rolled back and should be no longer available

Reply:
Rollback doesn't drop the transaction. It rolls it back to the initial state so you can continue to use it.